src/modules/75-cf/pages/onboarding/views/CreateAFlagView.tsx

```diff
@@ -5,79 +5,150 @@
  * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
  */
 
-import React, { useEffect } from 'react'
-import { Container, TextInput, Heading, Layout, Text } from '@wings-software/uicore'
-import { Color } from '@harness/design-system'
-import { String, useStrings } from 'framework/strings'
-import type { FeatureFlagRequestRequestBody } from 'services/cf'
+import React, { useEffect, useState } from 'react'
+import { Container, Layout, Text, Select } from '@harness/uicore'
+import { Color, FontVariation } from '@harness/design-system'
+import { useParams } from 'react-router-dom'
+import { Spinner } from '@blueprintjs/core'
+import { useStrings } from 'framework/strings'
+import {
+  CreateFeatureFlagQueryParams,
+  Feature,
+  FeatureFlagRequestRequestBody,
+  useCreateFeatureFlag,
+  useGetAllFeatures
+} from 'services/cf'
+import { CF_DEFAULT_PAGE_SIZE, getErrorMessage } from '@cf/utils/CFUtils'
 import { useTelemetry } from '@common/hooks/useTelemetry'
+import { useToaster } from '@common/exports'
 import { Category, FeatureActions } from '@common/constants/TrackingConstants'
 import { CreateAFlagInfoView } from './CreateAFlagInfoView'
-import css from './CreateAFlagView.module.scss'
-
+import { OnboardingSelectedFlag } from '../OnboardingSelectedFlag'
 export interface CreateAFlagViewProps {
-  setFlagName: React.Dispatch<React.SetStateAction<string>>
-  flagInfo: FeatureFlagRequestRequestBody
-  isCreated: boolean
-  goNext: () => void
+  selectedFlag?: Feature
+  setSelectedFlag: (flag?: Feature) => void
 }
 
-export const CreateAFlagView: React.FC<CreateAFlagViewProps> = ({ flagInfo, setFlagName, isCreated, goNext }) => {
+export const CreateAFlagView: React.FC<CreateAFlagViewProps> = ({ selectedFlag, setSelectedFlag }) => {
+  const { showError } = useToaster()
   const { getString } = useStrings()
-
   const { trackEvent } = useTelemetry()
+  const [searchTerm, setSearchTerm] = useState<string>('')
+  const [flagCreated, setFlagCreated] = useState(false)
+  const { orgIdentifier, accountId: accountIdentifier, projectIdentifier } = useParams<Record<string, string>>()
+
+  const { mutate: createFeatureFlag, loading: isLoadingCreateFeatureFlag } = useCreateFeatureFlag({
+    queryParams: {
+      accountIdentifier,
+      orgIdentifier
+    } as CreateFeatureFlagQueryParams
+  })
+
+  const queryParams = {
+    projectIdentifier,
+    accountIdentifier,
+    orgIdentifier,
+    name: searchTerm,
+    pageSize: CF_DEFAULT_PAGE_SIZE
+  }
+
+  const { data: allFeatureFlags, refetch: fetchFlags } = useGetAllFeatures({
```

Comment:
Absolute nitpick, but just to improve the readability of the code

```suggestion
  const { data: allFeatureFlags, refetch: refetchFlags } = useGetAllFeatures({
```